### ISSUE TITLE
Incorrect versions in some bundler outputs

### ISSUE THREAD

**Issue by sellout** [NixOS/bundlers#10]
It looks like some bundlers are dropping the leading `0.` for pre-1.0 versions.

Here it is for a project of mine – the `toDEB` and `toRPM` results both say the version is `1.0`, while `toDockerImage` has the correct `0.1.0` version.

```bash
$ nix bundle --bundler github:NixOS/bundlers#toDEB
$ nix bundle --bundler github:NixOS/bundlers#toDockerImage
$ nix bundle --bundler github:NixOS/bundlers#toRPM
$ ls *bash-strict-mode*
bash-strict-mode-0.1.0.tar.gz

deb-single-bash-strict-mode/:
.  ..  bash-strict-mode_1.0_amd64.deb

rpm-single-bash-strict-mode/:
.  ..  bash-strict-mode-1.0-1.x86_64.rpm
$ 
```

**Comment by dpc** [comment]
How do I control the versions on the deb and rpm bundlers anyway?

I have a bunch of binaries that get correct versions when the normal wrapped binaries are produced, but rpm and deb just put `1.0`: https://github.com/fedimint/fedimint/releases/tag/v0.3.0-rc.2

**Comment by sellout** [comment]
Ah! That might be the _actual_ problem – not that the leading component is dropped, but that RPMs and DEBs always get 1.0.

**Comment by mjakobsche** [comment]
I'm working on a fix for this https://github.com/juliosueiras-nix/nix-utils/pull/2

**Comment by mjakobsche** [comment]
since https://github.com/juliosueiras-nix/nix-utils/pull/2 is now merged it should work properly now

**Comment by dpc** [comment]
Can anyone verify? I think I have the newest version, and at least the filenames I'm getting have 1.0 version in them.

**Comment by mjakobsche** [comment]
Yes, sorry for misinformation, flake.lock update by @colorodoxyz must be merged here first in order for it to work properly

**Comment by dpc** [comment]
Oh. Right. Because how `nix bundle` is invoked. I thought `nix flake update bundlers/nix-utils` will do it, as my `flake.lock` did change.

**Comment by dpc** [comment]
Hmm... I updated bundlers inputs  to one containing #25 ... and I'm still getting 1.0 in the filenames. Anyone can verify?

**Comment by mjakobsche** [comment]
Hmm, it does work well for me now. Did you run command with refresh? 
`nix bundle --bundler github:NixOS/bundlers#toDEB --refresh`

**Comment by dpc** [comment]
Not working here. Tried `--refresh`, tried changing the version and causing a rebuild. I use `lib.traceVal` on the `version` of the output I'm bundling and it shows `0.9.0-alpha` (I also tried `0.9.0`).

I wonder where exactly is the version being sourced from. Our outputs are a bit special, because they pick binaries from a larger Rust workspace build. Do you think this should be OK:

```
  pickBinary =
    {
      pkg,
      name ? null,
      bin ? null,
      bins ? [ ],
    }:
    pkgs.stdenv.mkDerivation {
      system = pkgs.system;
      name = if name != null then name else bin;
      pname = if name != null then name else bin;
      version = lib.traceVal pkg.version;

      dontUnpack = true;
      # just don't mess with it
      dontStrip = true;

      installPhase = ''
        mkdir -p $out/bin
        ${if bin != null then "cp -a ${pkg}/bin/${bin} $out/bin/${bin}" else ""}
        ${builtins.concatStringsSep "\n" (map (b: "cp -a ${pkg}/bin/${b} $out/bin/${b}") bins)}
      '';
    };
```

?

**Comment by mjakobsche** [comment]
the version is determined by derivations ending part. if it cannot be parsed it will fallback to 1.0:
```nix
derivation = parseDrvName (
  elemAt (split "/" (elemAt (split "/[0-9a-df-np-sv-z]{32}-" (program)) 2)) 0
);
```
https://github.com/juliosueiras-nix/nix-utils/pull/2/commits/a16bbea9602174040731cd587310c723c1976e7e

**Comment by dpc** [comment]
Thanks! That was it. In my derivation above, I'm just not supposed to set `name`, so the output path automatically is `${pname}-${version}`. That fixes it.
